src/accelerate/test_utils/other.py

```diff
@@ -0,0 +1,118 @@
+# Copyright 2023 The HuggingFace Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import importlib
+import os
+from pathlib import Path
+from typing import Callable, Dict
+
+import torch
+
+
+# Set a backend environment variable for any extra module import required for a custom accelerator
+if "ACCELERATE_TEST_BACKEND" in os.environ:
+    backend = os.environ["ACCELERATE_TEST_BACKEND"]
+    try:
+        _ = importlib.import_module(backend)
+    except ModuleNotFoundError as e:
+        raise ModuleNotFoundError(
+            f"Failed to import `ACCELERATE_TEST_BACKEND` '{backend}'! This should be the name of an installed module \
+                to enable a specified backend.):\n{e}"
+        ) from e
+
+if "ACCELERATE_TEST_DEVICE" in os.environ:
+    torch_device = os.environ["ACCELERATE_TEST_DEVICE"]
+    try:
+        # try creating device to see if provided device is valid
+        _ = torch.device(torch_device)
+    except RuntimeError as e:
+        raise RuntimeError(
+            f"Unknown testing device specified by environment variable `ACCELERATE_TEST_DEVICE`: {torch_device}"
+        ) from e
+else:
+    torch_device = "cuda" if torch.cuda.is_available() else "cpu"
+
+
+# Utils for custom and alternative accelerator devices
+
+# Mappings from device names to callable functions to support device agnostic testing.
+BACKEND_DEVICE_COUNT = {"cuda": torch.cuda.device_count, "cpu": lambda: 0, "default": lambda: 1}
+BACKEND_IS_AVAILABLE = {"cuda": torch.cuda.is_available, "cpu": lambda: False, "default": torch.cuda.is_available}
+
+
+# This dispatches a defined function according to the accelerator from the function definitions.
+def _device_agnostic_dispatch(device: str, dispatch_table: Dict[str, Callable], *args, **kwargs):
```

Comment:
```suggestion
def device_agnostic_dispatch(device: str, dispatch_table: Dict[str, Callable], *args, **kwargs):
```

We tend not to have any private functions in accelerate, so let's keep this public even if not used :)

Reply:
done!